addons/sms/wizard/sms_composer.py

```diff
@@ -238,8 +238,9 @@ def _action_send_sms_mass(self, records=None):
         sms_all = self._prepare_mass_sms(records, sms_record_values)
 
         if sms_all and self.mass_keep_log and records and issubclass(type(records), self.pool['mail.thread']):
-            log_values = self._prepare_mass_log_values(records, sms_record_values)
-            records._message_log_batch(**log_values)
+            filtered_records = records.filtered(lambda r: r.id in sms_record_values)
```

Comment:
We can just browse then :)

`records = records.browse(list(sms_record_values.keys()))`